test/test_distillation.py

```diff
@@ -18,16 +26,36 @@ def test_distillation():
     student.distil_from(teacher, data_dir="samples/squad", train_filename="tiny.json")
 
     # create new checkpoint
-    new_student_weights = [torch.clone(param) for param in student.inferencer.model.parameters()]
-
-    new_student_weights = []
-    for name, weight in student.inferencer.model.named_parameters():
-        if "weight" in name and weight.requires_grad:
-            new_student_weights.append(weight)
+    new_student_weights = create_checkpoint(student)
 
     assert len(new_student_weights) == 22
 
     new_student_weights.pop(-2) # pooler is not updated due to different attention head
 
     # check if weights have changed
-    assert not any(torch.equal(old_weight, new_weight) for old_weight, new_weight in zip(student_weights, new_student_weights))
\ No newline at end of file
+    assert_weight_change(student_weights, new_student_weights)
+
+def test_tinybert_distillation():
+    student = FARMReader(model_name_or_path="huawei-noah/TinyBERT_General_4L_312D")
+    teacher = FARMReader(model_name_or_path="bert-base-uncased")
```

Comment:
This would be theoretically possible, but the teacher model would need to have the exact same dimensions except for the number of layers which would need to be a multiple of the number of student layers. This means it is quite hard to find a matching model. If it is a big performance issue we could perhaps create our own "mock model" with the right parameters.

Reply:
Alright, let's keep it as it is for now. 👍